src/main/java/microsoft/exchange/webservices/data/TimeZoneDefinition.java

```diff
@@ -149,7 +136,7 @@ protected void readAttributesFromXml(EwsServiceXmlReader reader)
     if (this.id == null || this.id.isEmpty()) {
       String nameValue = (this.getName() == null || this.
           getName().isEmpty()) ? "" : this.getName();
-      this.setId(NoIdPrefix + Math.abs(nameValue.hashCode()));
+      this.setId(NoIdPrefix + Integer.toString(nameValue.hashCode()).replace("-", ""));
```

Comment:
Math.abs(Integer.MAX) does not work. this workaround does not change the workings, but is quite ugly :-)